src/components/Account/Templates/Templates.vue

```diff
@@ -33,32 +33,49 @@ export default {
         Spinner
     },
 
+    props: {
+        allowEdit: {
+            type: Boolean,
+            default: true
+        },
+        allowAdd: {
+            type: Boolean,
+            default: true
+        },
+        allowDelete: {
+            type: Boolean,
+            default: true
+        },
+        setTitle: {
+            type: Boolean,
+            default: true
+        }
+    },
+
     data () {
         return {
-            title: "Templates",
             loading: true,
             templates: [],
+            selectedTemplate: null
         };
     },
 
+    watch: {
+        'selectedTemplate' () {
+            this.$emit('selected-template-text', this.selectedTemplate.text);
+        }
+    },
```

Comment:
There is no need for `watch` here. Just emit when `click` is triggered and `selectedTemplate` is actually set. It doesn't get set anywhere else.
```suggestion
```